cmake-modules/AzureTransportAdapters.cmake

```diff
@@ -15,6 +15,10 @@ endif()
 
 # On Windows: Make sure to build WinHTTP either if it was user-requested or no transport was selected at all.
 # On POSIX: Make sure to build Curl either if it was user-requested or no transport was selected at all.
+if (NO_AUTOMATIC_TRANSPORT_BUILD)
```

Comment:
Can we try to fix the issue without additional build flags like this? We should be careful and conservative introducing more flags, since it introduces permutation explosion of testing and documentation.

I don't see why the customer would want to set this flag (or when to turn it off).

Reply:
Package managers want to set this. Package managers don't want that projects are smart and enable stuff that was explicitly disabled.

Reply:
This option does not document the build flag, so the complexity for the customer to the customer is exactly the same as before.

I can't think of ways that would not introduce the build flags. If I knew how to fix it without the new flag, I would have made it that way. If you have a solution in mind, please propose one.

The only thing I can think of, that is somewhat different but still introduces the new flag, is adding a build flag called `BUILD_TRANSPORT_AUTO` with `ON` being the default, but that has more potential for a breaking change and need to be done correctly given that now you can only load a part of the SDK and build it, so updating only the root cmakefile is probably not enough.

BTW, it used to work correctly in vcpkg (using `AZ_ALL_LIBRARIES`), but it was broken in one of the PRs: https://github.com/Azure/azure-sdk-for-cpp/pull/3230/files
At a time, `AZ_ALL_LIBRARIES` would have been set when you build the entire SDK, which was the only entry point when you clone the repo, and only vcpkg would ever build libraries individually. Now that is no longer the case, so we need another way to disable the auto-transport logic.